src/D2L.CodeStyle.Analyzers/Immutability/TypeDeclarationImmutabilityAnalyzer.cs

```diff
@@ -110,5 +119,12 @@ INamedTypeSymbol analyzedType
 
 			return false;
 		}
+
+		private static bool IsFromGeneratedResourcesFile( ISymbol symbol ) {
+			var tree = symbol.DeclaringSyntaxReferences.Single().SyntaxTree;

```

Comment:
`Single()` -- I don't know how this could break for properties/fields.
It doesn't break in our codebase.